### DISCUSSION TITLE
Convert TagLab annotations into YOLO or COCO format

### DISCUSSION THREAD

**Discussion by andieich** [cnr-isti-vclab/TagLab#137] [General]
Is there a way to convert the TagLab annotations to YOLO or COCO format? I'd like to train my own model but have photo transect data, therefore I don't want to split the data as it seems to be the only possibility in TagLab...  
Just as a background: I have permanent photo quadrats for a couple of years and want to track individual colonies over time. Manually, this works quite well, but the automatic segmentation with the TagLab models doesn't seem to work, because they are made for orthomosaiks. I want to use the annotations I've done so far to train a YOLO model to detect the colonies and then use their masks to do the matching between the colonies at different time points manually in TagLab.

**Comment by Jordan-Pierce** [comment]
To clarify, you have photo quadrat images, but have they been annotated with segments in / outside of TagLab?

You can technically annotate images in TagLab, you just treat them as mosaics (maps). You can also export the segments in COCO format, it's an export option already available.

If you're training a YOLO model (ultralytics?) then you can also look at roboflow's supervision, which has a from_coco function if you need to convert to YOLO format.

**Reply by andieich** [reply]
Thanks for your help. I want to train a model outside of TagLab to be more flexible and just use it for the annotations. I'm struggling to export the annotations,  I get this error (I'm on a Mac M1 Pro):
```
 File "/Users/andi/Documents/PhD/taglab/TagLab-main/TagLab.py", line 4034, in exportNewDataset
    new_dataset.createLabelImage(self.project.labels)
  File "/Users/andi/Documents/PhD/taglab/TagLab-main/source/NewDataset.py", line 554, in createLabelImage
    fillPoly(self.id_image, pts=[points], color = blob.id)
cv2.error: OpenCV(4.9.0) :-1: error: (-5:Bad argument) in function 'fillPoly'
> Overload resolution failed:
>  - Layout of the output array img is incompatible with cv::Mat
>  - Expected Ptr<cv::UMat> for argument 'img'
zsh: abort      python TagLab.py
```
